screens/HomeScreen.js

```diff
@@ -38,11 +39,72 @@ export default class HomeScreen extends React.Component {
     header: null
   };
 
-  selectLandmark = data => {
+  fetchLandmarkPlaceId(address) {
+    const formattedLandmarkAddress = this.formatLandmarkText(address);
+
+    return fetch(
+      `https://maps.googleapis.com/maps/api/place/findplacefromtext/json?input=${formattedLandmarkAddress}&inputtype=textquery&key=${Constants.google.apiKey}`
+    )
+      .then(res => {
+        return res.json();
+      })
+      .then(data => {
+        //grab the first result that the search returns and
+        //get its place id
+        if (data.candidates[0]) {
+          return data.candidates[0].place_id;
+        } else {
+          return "none";
+        }
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  fetchLandmarkDetails(googlePlaceId) {
+    return fetch(
+      `https://maps.googleapis.com/maps/api/place/details/json?place_id=${googlePlaceId}&fields=opening_hours,formatted_phone_number,formatted_address&key=${Constants.google.apiKey}`
+    )
+      .then(responseDetails => {
+        return responseDetails.json();
+      })
+      .then(placeDetails => {
+        return placeDetails.result;
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  formatLandmarkText(text) {
+    let words = text.split(" ");
+    return words.join("%20");
+  }
+
+  async selectLandmark(data) {
+    let placeId = data.placeId;
+    let selectedLandmark = { ...data };
+
+    if (!placeId) {
+      //Fail safe for if we don't have the placeId in the db
+      placeId = await this.fetchLandmarkPlaceId(data.location);
+    }
+
+    //only check for placeDetails, if a placeId exists
+    //and we haven't already determined there is none
+    if (placeId !== "none") {
```

Comment:
when does placeId = "none"?

Reply:
We have a few landmarks that don't have google placeIds because they are not official places. I think we have three right now in our current data set but I have a feeling we'll probably have more just from the nature of what we're doing